default/AI/AIstate.py

```diff
@@ -1091,3 +1093,20 @@ def split_new_fleets(self): # pylint: disable=invalid-name
                 print "\t from splitting fleet ID %4d with %d ships, got %d new fleets:" % (fleet_id, fleet_len, len(new_fleets))
                 # old fleet may have different role after split, later will be again identified
                 #self.remove_fleet_role(fleet_id)  # in current system, orig new fleet will not yet have been assigned a role
+
+    def log_peace_request(self, initiating_empire_id, recipient_empire_id):
+        """Keep a record of peace requests made or received by this empire."""
+
+        peace_requests = self.diplomatic_logs.setdefault('peace_requests', {})
+        log_index = (initiating_empire_id, recipient_empire_id)
+        peace_requests.setdefault(log_index, []).append(fo.currentTurn())
+
+    def log_war_declarations(self, initiating_empire_id, recipient_empire_id):
+        """Keep a record of war declarations made or received by this empire."""
+
+        # if war declaration is made on turn 1, don't hold it against them
+        if fo.currentTurn() == 1:
+            return
+        war_declarations = self.diplomatic_logs.setdefault('war_declarations', {})
+        log_index = (initiating_empire_id, recipient_empire_id)
```

Comment:
Flat `list<tuple>` is more simple then `dict<str, dict<tuple, list>>`

`filter`function  and list comprehension allow to get any information in simple way.